MENTAL-MODEL.md

```diff
@@ -0,0 +1,376 @@
+    .send()
+    .with_context(|| "HTTP request failed")?;
+```
+
+### 6. Process the response
+
+An `Ok` response should include a Payjoin Proposal PSBT. Check that it's signed, following protocol, not trying to steal or otherwise error.
+
+```rs
+// TODO display well-known errors and log::debug the rest
+let psbt = ctx.process_response(response).with_context(|| "Failed to process response")?;
+log::debug!("Proposed psbt: {:#?}", psbt);
+```
+
+Payjoin response errors (called [receiver's errors in spec](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#receivers-well-known-errors)) come from a remote server and can be used to "maliciously to phish a non technical user." Only those listed as "well known" in the spec should be displayed with preset messages to prevent phishing.
+
+### 7. Sign and finalize the Payjoin Proposal PSBT
+
+Most software can handle adding the last signatures to a PSBT without issue.
+
+```rs
+let psbt = bitcoind
+    .wallet_process_psbt(&serialize_psbt(&psbt), None, None, None)
+    .with_context(|| "Failed to process PSBT")?
+    .psbt;
+let tx = bitcoind
+    .finalize_psbt(&psbt, Some(true))
+    .with_context(|| "Failed to finalize PSBT")?
+    .hex
+    .ok_or_else(|| anyhow!("Incomplete PSBT"))?;
+```
+
+### 8. Broadcast the Payjoin Transaction
+
+In order to preserve privacy between the transaction and the IP address from which it originates, transaction broadcasting should be done using Tor, a VPN, or proxy.
+
+```rs
+let txid =
+    bitcoind.send_raw_transaction(&tx).with_context(|| "Failed to send raw transaction")?;
+log::info!("Transaction sent: {}", txid);
+```
+
+📤 Sending payjoin is just that simple.
+
+## Receive a Payjoin
+
+The `receiver` feature provides all of the check methods, PSBT data manipulation, coin selection, and transport structures to receive payjoin and handle errors in a privacy preserving way.
+
+Receiving payjoin entails listening to a secure http endpoint for inbound requests. The endpoint is displayed in the `pj` parameter of a bip21 request URI.
+
+The reference implementation uses `rouille` sync http server and Bitcoin Core RPC.
+
+```rs
+fn receive_payjoin(
+    bitcoind: bitcoincore_rpc::Client,
+    amount_arg: &str,
+    endpoint_arg: &str,
+) -> Result<()>
+```
+
+### 1. Generate a pj_uri BIP21 using `payjoin::Uri::from_str`
+
+A BIP 21 URI supporting payjoin contains at minimum a bitcoin address and a secure `pj` endpoint.
+
+```rs
+let pj_receiver_address = bitcoind.get_new_address(None, None)?;
+let amount = Amount::from_sat(amount_arg.parse()?);
+let pj_uri_string = format!(
+    "{}?amount={}&pj={}",
+    pj_receiver_address.to_qr_uri(),
+    amount.to_btc(),
+    endpoint_arg
+);
+let pj_uri = Uri::from_str(&pj_uri_string)
+    .map_err(|e| anyhow!("Constructed a bad URI string from args: {}", e))?;
+let _pj_uri = pj_uri
+    .check_pj_supported()
+    .map_err(|e| anyhow!("Constructed URI does not support payjoin: {}", e))?;
+```
```

Comment:
A URI builder may be a more ergonomic interface. The `bip21` crate might be a good candidate to start to explore bindings with. It doesn't make sense to construct a string and then parse it rather than provide a concrete `PjUri` constructor taking `address: Address`, `amount: Option<amount>`, `pj: Endpoint`